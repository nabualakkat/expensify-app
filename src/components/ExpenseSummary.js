import React from 'react'
import {connect} from 'react-redux'
import numeral from 'numeral'
import selectExpensesTotal from '../selectors/expense-total'
import selectExpenses from '../selectors/expenses'

export const ExpenseSummary = (props) => {
    const expenseCount = props.expenses.length
    const expensesTotal = selectExpensesTotal(props.expenses)
    return (
        <div>
            <p>Viewing {expenseCount} {expenseCount===1?"expense":"expenses"} totalling {numeral(expensesTotal / 100).format("$0,0.00")}</p>
        </div>
    )
}

const mapStateToProps = (state) => {
    return{
        expenses: selectExpenses(state.expenses, state.filters)
    }
}

export default connect(mapStateToProps)(ExpenseSummary)