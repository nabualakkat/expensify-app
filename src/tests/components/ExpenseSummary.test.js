import React from 'react'
import {shallow} from 'enzyme'
import {ExpenseSummary} from '../../components/ExpenseSummary'
import expenses from '../fixtures/expenses'

test('should render ExpenseSummary with plural expenses', () => {
    const wrapper = shallow(<ExpenseSummary expenses={expenses}/>)
    expect(wrapper).toMatchSnapshot()
})

test('should render ExpenseSummary with singular expense', () => {
    const wrapper = shallow(<ExpenseSummary expenses={[expenses[0]]}/>)
    expect(wrapper).toMatchSnapshot()
})