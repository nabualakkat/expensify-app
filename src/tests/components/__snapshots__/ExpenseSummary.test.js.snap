// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseSummary with plural expenses 1`] = `
<div>
  <p>
    Viewing 
    3
     
    expenses
     totalling 
    $120.95
  </p>
</div>
`;
